src/stories/Tab.stories.tsx

```diff
@@ -0,0 +1,50 @@
+import type { Meta, StoryObj } from '@storybook/react';
+import { useState } from 'react';
+
+import { ReactComponent as CameraIcon } from '@/assets/camera.svg';
+import { Tab } from '@/domain/_common/components';
+
+const meta = {
+  title: 'Components/Tab',
+  component: Tab,
+  parameters: {
+    layout: 'centered',
+  },
+  tags: ['autodocs'],
+} satisfies Meta<typeof Tab>;
+
+export default meta;
+type Story = StoryObj<typeof meta>;
+
+export const Default: Story = {
+  render: () => {
+    const [selected, setSelected] = useState(1);
+
+    return (
+      <div className="bg-gray-100 p-10">
+        <Tab>
+          {[1, 2, 3].map((number) => (
+            <Tab.Item
+              key={number}
+              icon={<CameraIcon />}
+              text={`${number}해라`}
+              onClick={() => {
+                setSelected(number);
+              }}
+              selected={selected === number}
+            />
+          ))}
+        </Tab>
+      </div>
+    );
+  },
```

Comment:
아이디어를 드릴게용 (꼭 이렇게 해달라는건 아니고 참고만..!)
1. selectedIndex state는 Tab 내부 컨텍스트에서 정의
2. `Tab`
- 최초에 선택되어 있는 탭의 index를 가리키는 defautIndex 가 필요할 것 같음 + 이 값을 ContextProvider에서 넘겨줄 selectedIndex의 초기값으로 주면 될듯
3. `Tab.Item`
- 선택된 tab이 달라질 때 탭 외부에서 무언가 조작이 가능하도록 index를 파라미터로 받는 onClick 핸들러 달기
- 각 요소에 index prop을 항상 넘겨줘서, `Tab.Item` 내부에서 `selectedIndex === index` 로 `selected` 값 계산해서 사용

```suggestion
  render: () => (
      <Tab defaultIndex={1}>
        {[1, 2, 3].map((number, index) => (
          <Tab.Item
            key={number}
            index={index}
            icon={<CameraIcon />}
            text={`${number}해라`}
            onClick={index => console.log(index)}
          />
        ))}
      </Tab>
    </div>
  )
```

Reply:
그리고 만약 위와같이 onClick이 달리려면 TabItem의 내부에는
```jsx
  const handleClick = () => {
    context.setSelectedIndex(props.index);
    props.onClick?.(props.index);
  };
 ```
 이런 핸들러가 달려있겠네용